File: src/main/java/com/FrameHopper/app/API/Mapper/CommentMapper.java
package com.FrameHopper.app.API.Mapper;

import com.FrameHopper.app.API.Model.CommentDTO;
import com.FrameHopper.app.Model.Comment;
import org.hibernate.LazyInitializationException;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class CommentMapper {
    public static List<CommentDTO> mapComments(List<Comment> comments) {
        try {
            if(comments == null || comments.isEmpty())
                return new ArrayList<>();

            return comments.stream()
                    .sorted(Comparator.comparingInt(Comment::getListingOrder))
                    .map(CommentDTO::new)
                    .toList();
        } catch (LazyInitializationException e) {
            return new ArrayList<>();
        }
    }
}

File: src/main/java/com/FrameHopper/app/adapters/api/mappers/CommentMapper.kt
package com.FrameHopper.app.adapters.api.mappers

import com.FrameHopper.app.adapters.api.model.CommentExposureDTO
import com.FrameHopper.app.boundry.dto.CommentDTO

object CommentMapper {
    fun toExposure(comment: CommentDTO): CommentExposureDTO {
        return CommentExposureDTO(comment.id, comment.content)
    }
}